libs/community/langchain_community/embeddings/baichuan.py

```diff
@@ -78,26 +85,35 @@ def _embed(self, texts: List[str]) -> Optional[List[List[float]]]:
             A list of list of floats representing the embeddings, or None if an
             error occurs.
         """
-        response = self.session.post(
-            BAICHUAN_API_URL, json={"input": texts, "model": self.model_name}
-        )
-        # Raise exception if response status code from 400 to 600
-        response.raise_for_status()
-        # Check if the response status code indicates success
-        if response.status_code == 200:
-            resp = response.json()
-            embeddings = resp.get("data", [])
-            # Sort resulting embeddings by index
-            sorted_embeddings = sorted(embeddings, key=lambda e: e.get("index", 0))
-            # Return just the embeddings
-            return [result.get("embedding", []) for result in sorted_embeddings]
-        else:
-            # Log error or handle unsuccessful response appropriately
-            # Handle 100 <= status_code < 400, not include 200
-            raise RequestException(
-                f"Error: Received status code {response.status_code} from "
-                "`BaichuanEmbedding` API"
+        chunk_texts = [
+            texts[i : i + self.chunk_size]
+            for i in range(0, len(texts), self.chunk_size)
+        ]
+        embed_results = []
+        for chunk in chunk_texts:
+            response = self.session.post(
+                BAICHUAN_API_URL, json={"input": chunk, "model": self.model_name}
             )
+            # Raise exception if response status code from 400 to 600
+            response.raise_for_status()
+            # Check if the response status code indicates success
+            if response.status_code == 200:
+                resp = response.json()
+                embeddings = resp.get("data", [])
+                # Sort resulting embeddings by index
+                sorted_embeddings = sorted(embeddings, key=lambda e: e.get("index", 0))
```

Comment:
Previously, the embeddings were "globally" sorted. Here they are sorted within each chunk. What impact could this have?

Reply:
I don't think so.
Before:
The maximum number of documents passed in is 16 (any more will result in an error), and sorting is performed for embeddings of less than or equal to 16 documents.
Now:
The sorting of the first 16 documents is also completed, and there is no problem with the extra documents (17-more) being sorted later.